src/dash-playlist-loader.js

```diff
@@ -140,7 +140,8 @@ export const updateMaster = (oldMaster, newMaster, sidxMapping) => {
     if (playlist.sidx) {
       const sidxKey = generateSidxKey(playlist.sidx);
 
-      if (sidxMapping && sidxMapping[sidxKey]) {
+      // add sidx segments to the playlist if we have all the sidx info already
+      if (sidxMapping && sidxMapping[sidxKey] && sidxMapping[sidxKey].sidx) {
```

Comment:
in `addSidxSegmentsToPlaylist` we try to use `sidxMappind[sidxKey].sidx` but it can be undefined if the request for sidx and parsing is still in progress (at which point it will be added). So we need to add the additional key check here.